### PR TITLE
chore: Groups common code in data source and resource for populating root fields

### PR DESCRIPTION
**Pull request by maastha**
## Description

Groups common code in data source and resource for populating root fields.

Link to any related issue(s): [CLOUDP-259838](https://jira.mongodb.org/browse/CLOUDP-259838)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/advancedcluster/resource_advanced_cluster.go [hunk 1]
```diff
@@ -650,7 +649,7 @@ func setResourceRootFields(d *schema.ResourceData, cluster *admin.ClusterDescrip
 		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "version_release_system", clusterName, err))
 	}
 
-	if err := d.Set("accept_data_risks_and_force_replica_set_reconfig", conversion.TimePtrToStringPtr(cluster.AcceptDataRisksAndForceReplicaSetReconfig)); err != nil {
+	if err := d.Set("accept_data_risks_and_force_replica_set_reconfig", conversion.TimePtrToStringPtr(cluster.AcceptDataRisksAndForceReplicaSetReconfig)); isResourceSchema && err != nil {
```
Comment (EspenAlbert) [inline internal/service/advancedcluster/resource_advanced_cluster.go]: should isResourceSchema be before d.Set()? 
Or we are simply ignoring the error from d.Set when isResourceSchema is set? ( I would prefer avoiding the error in the first place)

**Review by EspenAlbert — APPROVED** [review]

File: internal/service/advancedcluster/resource_advanced_cluster.go (hunk 1 above)
Comment (maastha) [reply internal/service/advancedcluster/resource_advanced_cluster.go]: ya currently it just ignores the error, no strong opinion, I'll update it

**Review by lantoli — APPROVED** [review]

File: internal/service/advancedcluster/resource_advanced_cluster.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster.go]: I do see that `accept_data_risks_and_force_replica_set_reconfig` and `cluster_id` are not part of the data source schema, not sure if there is a risk in setting the values if not defined in the schema.

Comment (maastha) [reply internal/service/advancedcluster/resource_advanced_cluster.go]: updated, they are only set if `isResourceSchema` is true

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Pull request merged**